### PR TITLE
Fix python 3 related warnings

### LINKED ISSUE #1487: Deprecation warning due to invalid escape sequences in Python 3.7
Deprecation warning due to invalid escape sequences in Python 3.7. These can be resolved using raw strings or escaping the sequences again. Also helpful fpr automated fixes : https://github.com/asottile/pyupgrade/

```
find . -iname '*py' | grep -Ev 'setup|rdf4' | xargs -P4 -I{} python3.8 -Wall -m py_compile {}
./pwnlib/util/sh_string.py:3: DeprecationWarning: invalid escape sequence \)
  """
./pwnlib/lexer.py:40: DeprecationWarning: invalid escape sequence \.
  identifier = r'(?:[a-zA-Z$_]' + char + '*|\.' + char + '+|or)'
./examples/clean_and_log.py:28: DeprecationWarning: invalid escape sequence \S
  print(re.findall('^prefix (\S+)$', line)[0])
```

### PR DESCRIPTION
**Pull request by tirkarthi**
Fixes #1487 
Fixees #1486

### CODE REVIEW

File: pwnlib/util/sh_string.py [hunk 1]
```diff
@@ -49,9 +49,9 @@
     Double Quotes
       Enclosing characters within double quotes preserves the literal meaning
       of all characters except dollarsign ($), backquote (`), and backslash
-      (\).  The backslash inside double quotes is historically weird, and
+      (\\).  The backslash inside double quotes is historically weird, and
       serves to quote only the following characters:
-            $ ` " \ <newline>.
+            $ ` " \\ <newline>.
```
Comment (zachriggle) [inline pwnlib/util/sh_string.py]: Can you elaborate on why this change is necessary?

Comment (Arusekk) [reply pwnlib/util/sh_string.py]: This should be done using `r"""` in line 3 instead, and leaving original content here and in other parts of this docstring.

**Review by Arusekk — CHANGES_REQUESTED** [review]

File: pwnlib/util/sh_string.py (hunk 1 above)
Comment (tirkarthi) [reply pwnlib/util/sh_string.py]: Added raw strings as suggested. Not sure why pyupgrade escaped them individually. I don't see any unicode characters in docstring affected by using raw literals.

**Comment by tirkarthi** [comment]
Thanks for the review and merge :)
